lib/src/main/java/io/cloudquery/internal/servers/plugin/v3/PluginServer.java

```diff
@@ -9,4 +25,100 @@ public class PluginServer extends PluginImplBase {
     public PluginServer(Plugin plugin) {
         this.plugin = plugin;
     }
+
+    @Override
+    public void getName(io.cloudquery.plugin.v3.GetName.Request request,
+            StreamObserver<io.cloudquery.plugin.v3.GetName.Response> responseObserver) {
+        responseObserver
+                .onNext(io.cloudquery.plugin.v3.GetName.Response.newBuilder().setName(plugin.getName()).build());
+        responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getVersion(io.cloudquery.plugin.v3.GetVersion.Request request,
+            StreamObserver<io.cloudquery.plugin.v3.GetVersion.Response> responseObserver) {
+        responseObserver.onNext(
+                io.cloudquery.plugin.v3.GetVersion.Response.newBuilder().setVersion(plugin.getVersion()).build());
+        responseObserver.onCompleted();
+    }
+
+    @Override
+    public void init(io.cloudquery.plugin.v3.Init.Request request,
+            StreamObserver<io.cloudquery.plugin.v3.Init.Response> responseObserver) {
+        plugin.init();
+        responseObserver.onNext(io.cloudquery.plugin.v3.Init.Response.newBuilder().build());
+        responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getTables(io.cloudquery.plugin.v3.GetTables.Request request,
+            StreamObserver<io.cloudquery.plugin.v3.GetTables.Response> responseObserver) {
+        try {
+            List<Table> tables = plugin.tables();
+            List<ByteString> byteStrings = new ArrayList<>();
+            for (Table table : tables) {
+                try (BufferAllocator bufferAllocator = new RootAllocator()) {
```

Comment:
The root allocator is implemented via https://github.com/apache/arrow/tree/6357c9f2419d5b0717e62adc8233c649e10de34b/java/memory/memory-netty/src/main/java/org/apache/arrow/memory